packages/shared/test-suites/RecordingTestMetadata.ts

```diff
@@ -572,8 +572,11 @@ export async function convertGroupedTestCases(
         const annotationsByTest: Annotation[][] = annotations.reduce(
           (accumulated: Annotation[][], annotation: Annotation) => {
             if (annotation.message.event === "test:start") {
+              // Start a new annotations array for each test
               accumulated.push([annotation]);
-            } else {
+            } else if (accumulated.length > 0) {
+              // Else add new annotations to the current test
+              // (Note some recordings have annotations data before the first "test:start")
```

Comment:
This is not related to the rest of the PR, but while testing this change, I noticed a recording that had several annotations in the data _before_ the first `"test:start"` event– which is a case the previous code did not anticipate or handle.